Makefile

```diff
@@ -55,18 +55,18 @@ CFLAGS += $(DEFS) -I. -I./ $(STM32_INCLUDES)
 CFLAGS += -fsingle-precision-constant -Wdouble-promotion
 CFLAGS += -DLUA_32BITS -DLUA_COMPAT_5_2
 CFLAGS += -fno-common
-
+CFLAGS += -DVERSION=\"$(GIT_VERSION)\"
```

Comment:
Yes i noticed that in the article you linked. Couldn't see a simple way around it, and was trying to avoid spending too long on this problem.

In practice I don't think it'll be an issue as I always `make clean` before building a release. The `make clean` is also necessary if you want to set the `RELEASE` define which stops the uart debugger from initializing.

Would be great to use a more sophisticated build system but I only have experience with Make & am focused on completing the 1.0 release. Help on improving this in the future would be amazing!